code/hud/hudconfig.cpp

```diff
@@ -1786,42 +1882,39 @@ void hud_config_process_colors()
 {	
 }
 
-void hud_config_color_init()
+void hud_config_preset_init()
 {
 	HC_current_file = -1;
-
-	// get a list of all hcf files
-	memset(HC_filenames, 0, sizeof(char*) * MAX_HCF_FILES);
-	HC_num_files = cf_get_file_list(MAX_HCF_FILES, HC_filenames, CF_TYPE_PLAYERS, "*.hcf", CF_SORT_NAME, nullptr,
-	                                CF_LOCATION_ROOT_USER | CF_LOCATION_ROOT_GAME | CF_LOCATION_TYPE_ROOT);
+	HC_preset_filenames.clear();
+	cf_get_file_list(HC_preset_filenames, CF_TYPE_PLAYERS, "*.hcf", CF_SORT_NAME, nullptr,
+						CF_LOCATION_ROOT_USER | CF_LOCATION_ROOT_GAME | CF_LOCATION_TYPE_ROOT);
 }
 
-void hud_config_color_close()
+void hud_config_delete_preset(SCP_string filename)
```

Comment:
Addendum: this should use `const SCP_string &`

Reply:
No, it shouldn't because we modify the string here

Reply:
I stand corrected.